### PR TITLE
Fix REST fixture clash

### PR DESCRIPTION
**Pull request by mfalesni**
@lehinevych - this might introduce a merge conflict but it should be pretty easy to resolve, it only involves renaming of couple of things.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 7b70d026eadc1d5f4ab0b5029c96d8a9f2ddbc72:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav This should also be good to merge. Only one failure in 5.5 from the known argumenterror, 5.4 again failed to provision an appliance.

File: cfme/tests/test_rest.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 
 @pytest.fixture(scope="module")
-def provider():
+def a_provider():
```
Comment (psav) [inline cfme/tests/test_rest.py]: We should maybe make a fixture for setup_an_infra and setup_a_cloud provider.....what do you think?

Comment (mfalesni) [inline cfme/tests/test_rest.py]: I would not call it setup_ ... I think just a_cloud_provider[_modscope], an_infra_provider, ...

**Pull request merged**